tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,385 @@
+    actual_decisions = [b.pop("decisions") for b in alg.metadata_["brackets"].values()]
+    paper_decisions = [b.pop("decisions") for b in alg.metadata()["brackets"].values()]
+
+    for paper_iter, actual_iter in zip(paper_decisions, actual_decisions):
+        trimmed_paper_iter = {k for k in paper_iter if k <= alg_patience}
+        assert trimmed_paper_iter.issubset(set(actual_iter))
+        assert all(x <= alg_patience + 1 for x in actual_iter)
+
+    assert alg.metadata_["partial_fit_calls"] <= alg.metadata()["partial_fit_calls"]
+
+    alg = HyperbandSearchCV(model, params, max_iter=max_iter, patience=1)
+    with pytest.warns(UserWarning, match="The goal of `patience`"):
+        yield alg.fit(X, y)
+
+
+@gen_cluster(client=True, timeout=5000)
+def test_integration(c, s, a, b):
+    X, y = make_classification(n_samples=10, n_features=4, chunks=10)
+    model = ConstantFunction()
+    params = {"value": scipy.stats.uniform(0, 1)}
+    alg = HyperbandSearchCV(model, params, max_iter=9, random_state=42)
+    yield alg.fit(X, y)
+    cv_res_keys = set(alg.cv_results_.keys())
+    gt_zero = lambda x: x >= 0
+    gt_one = lambda x: x >= 1
+    for column, dtype, condition in [
+        ("params", dict, lambda d: set(d.keys()) == {"value"}),
+        ("test_score", float, gt_zero),
+        ("test_score", float, gt_zero),
+        ("rank_test_score", int, gt_one),
+        ("mean_partial_fit_time", float, gt_zero),
+        ("std_partial_fit_time", float, gt_zero),
+        ("mean_score_time", float, gt_zero),
+        ("std_score_time", float, gt_zero),
+        ("estimator_id", None, lambda x: isinstance(x, str)),
+        ("partial_fit_calls", int, gt_zero),
+        ("param_value", float, gt_zero),
+    ]:
+        if dtype:
+            assert alg.cv_results_[column].dtype == dtype
+        if condition:
+            assert all(condition(x) for x in alg.cv_results_[column])
+        cv_res_keys -= {column}
+
+    # the keys listed in the for-loop are all the keys in cv_results_
+    assert cv_res_keys == set()
+
+    alg.best_estimator_.fit(X, y)
+    alg.best_estimator_.score(X, y)
+    alg.score(X, y)
+
+    # Test types/format of all parameters we set after fitting
+    assert isinstance(alg.best_index_, int)
+    assert isinstance(alg.best_estimator_, ConstantFunction)
+    assert isinstance(alg.best_score_, float)
+    assert isinstance(alg.best_params_, dict)
+    assert isinstance(alg.history_, list)
+    assert all(isinstance(h, dict) for h in alg.history_)
+    assert isinstance(alg.model_history_, dict)
+    assert all(vi in alg.history_ for v in alg.model_history_.values() for vi in v)
+    assert all(isinstance(v, np.ndarray) for v in alg.cv_results_.values())
+    assert isinstance(alg.multimetric_, bool)
+    assert all("bracket=" in h["estimator_id"] for h in alg.history_)
+
+    keys = {
+        "score",
+        "score_time",
+        "partial_fit_calls",
+        "partial_fit_time",
+        "estimator_id",
```

Comment:
This should be changed to `model_id`. It's the work of another PR to change to `estimator_id`.